packages/edit-post/src/components/header/index.js

```diff
@@ -136,35 +138,40 @@ function Header( {
 							ref={ blockToolbarRef }
 							name="block-toolbar"
 						/>
-						{ isEditingTemplate && hasBlockSelected && (
-							<Button
-								className="edit-post-header__block-tools-toggle"
-								icon={ isBlockToolsCollapsed ? next : previous }
-								onClick={ () => {
-									setIsBlockToolsCollapsed(
-										( collapsed ) => ! collapsed
-									);
-								} }
-								label={
-									isBlockToolsCollapsed
-										? __( 'Show block tools' )
-										: __( 'Hide block tools' )
-								}
-							/>
-						) }
+						{ ( isEditingTemplate || isEditingPattern ) &&
+							hasBlockSelected && (
+								<Button
+									className="edit-post-header__block-tools-toggle"
+									icon={
+										isBlockToolsCollapsed ? next : previous
+									}
+									onClick={ () => {
+										setIsBlockToolsCollapsed(
+											( collapsed ) => ! collapsed
+										);
+									} }
+									label={
+										isBlockToolsCollapsed
+											? __( 'Show block tools' )
+											: __( 'Hide block tools' )
+									}
+								/>
+							) }
 					</>
 				) }
 				<div
 					className={ classnames( 'edit-post-header__center', {
 						'is-collapsed':
-							isEditingTemplate &&
+							( isEditingTemplate || isEditingPattern ) &&
 							hasBlockSelected &&
 							! isBlockToolsCollapsed &&
 							hasFixedToolbar &&
 							isLargeViewport,
 					} ) }
 				>
-					{ isEditingTemplate && <DocumentBar /> }
+					{ ( isEditingTemplate || isEditingPattern ) && (
+						<DocumentBar />
+					) }
```

Comment:
I think this is one area where `post-only` mode won't work for patterns. I guess it should be dependent on post type for now.

Not sure if we want to keep some config like this to drive all these aspects of the editor:
```js
const DEFAULT_EDITOR_INTERFACE = {
  hasDocumentBar: false,
  hasSurround: false,
};
const POST_TYPE_EDITOR_INTERFACE = {
  wp_block: {
    hasDocumentBar: true,
    hasSurround: true,
  }
};
```